app/controllers/application_controller.rb

```diff
@@ -26,12 +26,27 @@ class ApplicationController < ActionController::Base
   before_action :set_account_specific_connections!
   before_action :elevate_single_tenant!, if: :singletenant?
 
+  after_action :clear_session_cookie
+
   rescue_from Apartment::TenantNotFound do
     raise ActionController::RoutingError, 'Not Found'
   end
 
   protected
 
+  def clear_session_cookie
+    return if current_user || !cached_route?(request.path)
+
+    if response.headers["Cache-Control"].blank? || !(response.headers["Cache-Control"].include?('no-store') || response.headers["Cache-Control"].include?('no-cache'))
+      # this skips sending a session cookie # (a session cookie will cause cloudflare to avoid caching it)
+      request.session_options[:skip] = true
+    end
+  end
+
+  def cached_route?(path)
+    path.start_with?('/concern', '/catalog')
```

Comment:
Form routes start with `/concern` (e.g. `/concern/generic_works/new`), but I think this should be fine since you also have `if current_user` in `#clear_session_cookies`

Reply:
also the "new page" isn't an issue because its just a blank form. the POST and PUT aren't ever cached